Lombiq.HelpfulLibraries/Libraries/Contents/ContentPartDefinitionBuilderExtensions.cs

```diff
@@ -0,0 +1,57 @@
+﻿using OrchardCore.ContentManagement.Metadata.Settings;
+using System;
+using System.Linq.Expressions;
+
+namespace OrchardCore.ContentManagement.Metadata.Builders
+{
+    public static class ContentPartDefinitionBuilderExtensions
+    {
+        /// <summary>
+        /// Wraps an existing <see cref="ContentPartDefinitionBuilder"/> into a wrapped type.
+        /// </summary>
+        public static ContentPartDefinitionBuilder<TPart> AsPart<TPart>(this ContentPartDefinitionBuilder builder)
+            where TPart : ContentPart =>
+            new(builder);
+    }
+
+    public class ContentPartDefinitionBuilder<TPart>
+        where TPart : ContentPart
+    {
+        public ContentPartDefinitionBuilder Builder { get; private set; }
+
+        public ContentPartDefinitionBuilder(ContentPartDefinitionBuilder builder) => Builder = builder;
+
+        public ContentPartDefinitionBuilder<TPart> Configure(Action<ContentPartDefinitionBuilder> configureAction)
+        {
+            configureAction(Builder);
+            return this;
+        }
+
+        /// <summary>
+        /// Creates a field by getting the technical name and field type using a lambda expression rather than relying
+        /// on developer provided strings. This improves the coupling between the part and the field. Also sets the
+        /// display name to the field's technical name by default, but if a more natural name is needed it can be
+        /// overwritten in the <paramref name="configuration"/>.
+        /// </summary>
+        /// <param name="fieldPropertySelector">The expression which gets the part and returns its field.</param>
+        /// <param name="configuration">Any further configuration, can override the automatic settings.</param>
+        /// <typeparam name="TField">The type of the new field to be attached to the content part.</typeparam>
+        public ContentPartDefinitionBuilder<TPart> WithField<TField>(
+            Expression<Func<TPart, TField>> fieldPropertySelector,
+            Action<ContentPartFieldDefinitionBuilder> configuration = null)
+        {
+            var property = ((MemberExpression)fieldPropertySelector.Body).Member;
+            var name = property.Name;
+
+            Builder = Builder.WithField(
+                name,
+                field =>
+                {
+                    field = field.WithDisplayName(name).OfType(typeof(TField).Name);
+                    configuration?.Invoke(field);
+                });
+
+            return this;
+        }
+    }
+}

```

Comment:
I've built something very similar for Form Field configuration in CARE 😁

Reply:
Great minds think alike? 😁

Is there something more along those lines there that I should add here? Now is as good a time as any to port them.

Reply:
I've looked at my implementation and cannot find anything else interesting. Maybe you could create a PR in SNOW for this issue so I can take a look at the usage of these - that might be inspiring.

Reply:
- https://github.com/Lombiq/Willis-Towers-Watson/pull/9/files#diff-a2944109dd84dffff19133d00ddd03cdb81a7aa9ad68ff0d38c99302f07728ac
- https://github.com/Lombiq/Helpful-Libraries/pull/109/files#diff-a71f9982acffec2f5f2e72caf2115413809156b67004b2c473e39a4dd90d1949